### PR TITLE
.Net: Marking the ToolCallResultSerializerOptions as obsolete.

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
The `ToolCallBehavior.ToolCallResultSerializerOptions` property will be replaced by an alternative solution soon.

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]

**Comment by stephentoub** [comment]
Thanks.

File: dotnet/src/Connectors/Connectors.OpenAI/ToolCallBehavior.cs [hunk 1]
```diff
@@ -91,6 +93,8 @@ private ToolCallBehavior(bool autoInvoke)
     /// <summary>
     /// Options to control tool call result serialization behavior.
     /// </summary>
+    [Obsolete("This property is deprecated in favor of Kernel.SerializerOptions that will be introduced in one of the following releases.")]
```
Comment (stephentoub) [inline dotnet/src/Connectors/Connectors.OpenAI/ToolCallBehavior.cs:96]: You'll probably need to suppress the warnings you get on the various use sites.

Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.OpenAI/ToolCallBehavior.cs:96]: Yep, suppressed.

**Review by markwallace-microsoft — APPROVED** [review]

File: dotnet/src/Connectors/Connectors.OpenAI/ToolCallBehavior.cs (hunk 1 above)
Comment (MediVik333) [reply dotnet/src/Connectors/Connectors.OpenAI/ToolCallBehavior.cs:96]: Ok

**Comment by MediVik333** [comment]
Ok

**Pull request merged**